main.go

```diff
@@ -52,8 +52,14 @@ type Module struct {
 	DNS     DNSProbe      `yaml:"dns"`
 }
 
+type Probe struct {
+	Protocol            string `yaml:"protocol"`
```

Comment:
This would be clearer as l4_protocol or transport_protocol, however the TCP and HTTP modules only support TCP - so this should be a setting only in the DNS module.